test/functional/api/buffer_spec.lua

```diff
@@ -117,36 +117,6 @@ describe('api/buf', function()
       eq({5, 2}, meths.win_get_cursor(win2))
     end)
 
-    it('line_count has defined behaviour for unloaded buffers', function()
```

Comment:
is it fine to remove these two specs? these were there to make sure there was no UB for unloaded buffers, but since this PR makes sure buffer related methods ensure that the buffer is loaded, these are unnecessary if I understand correctly 
(these specs were also failing, so I wondering if its even worth it to fix them)

Reply:
Tests should be kept but updated to new behavior. 

Also we need to document this as a breaking change. On master and before it is possible to use `nvim_buf_line_count` to both detect if a buffer is loaded or not (a loaded buffer never has zero lines) and how big it is if it's loaded. I don't think there are many plugins who do this anymore, but it needs to be clearly documented, so if a plugin breaks this can be found as a likely cause.